### PR TITLE
TF-2101 [Part 2] Add date selection fields to recover deleted message dialog

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
---
#2101 

### Need merge
---
https://github.com/linagora/tmail-flutter/pull/2406

## Resolved
---
- Web

https://github.com/linagora/tmail-flutter/assets/80142234/73894497-58da-4a7d-a555-b2cc163d8f87

- Android

https://github.com/linagora/tmail-flutter/assets/80142234/5f7a29a9-e719-4138-a871-ff0a1abe4ac7

### CODE REVIEW

**Review by hoangdat — APPROVED** [review]

File: lib/features/email_recovery/presentation/controller/input_field_focus_manager.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/material.dart';
+
+class InputFieldFocusManager {
+  late FocusNode deletionDateFieldFocusNode;
+  late FocusNode receptionDateFieldFocusNode;
+  late FocusNode subjectFieldFocusNode;
+  late FocusNode recipientsFieldFocusNode;
+  late FocusNode senderFieldFocusNode;
+  late FocusNode attachmentCheckboxFocusNode;
+  late FocusNode createButtonFocusNode;
```
Comment (dab246) [inline lib/features/email_recovery/presentation/controller/input_field_focus_manager.dart:10]: Why do we need to create so many FocusNodes?

File: lib/features/email_recovery/presentation/styles/date_selection_field/date_selection_dropdown_button_styles.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  static const double checkedIconSize = 20.0;
+
+  static const int dropdownElevation = 4;
+
+  static const Offset dropdownOffset = Offset(0.0, -8.0);
+
+  static const EdgeInsetsGeometry padding = EdgeInsets.only(
+    left: 12,
+    right: 10
+  );
```
Comment (dab246) [inline lib/features/email_recovery/presentation/styles/date_selection_field/date_selection_dropdown_button_styles.dart:18]: please use `EdgeInsetsDirectional`

File: lib/features/email_recovery/presentation/widgets/email_recovery_form/email_recovery_form_desktop_builder.dart [hunk 3]
```diff
@@ -27,6 +29,41 @@ class EmailRecoveryFormDesktopBuilder extends GetWidget<EmailRecoveryController>
                 style: EmailRecoveryFormStyles.titleTextStyle,
               ),
             ),
+            Expanded(
+              child: SingleChildScrollView(
+                physics: const ClampingScrollPhysics(),
+                child: Padding(
+                  padding: EmailRecoveryFormStyles.inputAreaPadding,
+                  child: FocusTraversalGroup(
```
Comment (dab246) [inline lib/features/email_recovery/presentation/widgets/email_recovery_form/email_recovery_form_desktop_builder.dart:37]: why we need `FocusTraversalGroup`?

**Comment by dab246** [comment]
@hieutbui  please rebase it

**Review by dab246 — APPROVED** [review]

**Pull request merged**